utils/utils.go

```diff
@@ -79,6 +79,15 @@ func CopyFile(src, dst string) error {
 		return err
 	}
 
+	fi, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	if err := os.Chmod(dst, fi.Mode()); err != nil {
```

Comment:
If you copy a key it will not have the correct permissions.  When you then try to ssh, it will fail because we shell out to ssh which the file must be `0600`.  Here I copy the original file permissions to match the existing key perms.